M/MUMPS_seq/build_tarballs.jl

```diff
@@ -0,0 +1,102 @@
+using BinaryBuilder
+
+name = "MUMPS_seq"
+version = v"5.2.1"
+
+sources = [
+  ArchiveSource("http://mumps.enseeiht.fr/MUMPS_5.2.1.tar.gz",
+                "d988fc34dfc8f5eee0533e361052a972aa69cc39ab193e7f987178d24981744a"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+mkdir -p ${libdir}
+cd $WORKSPACE/srcdir/MUMPS_5.2.1
+
+OPENBLAS=(-lopenblas)
+FFLAGS=()
+CFLAGS=()
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+  FFLAGS+=(-ffixed-line-length-none)  # replacing symbols below sometimes makes lines > 72 chars
+  OPENBLAS=(-lopenblas64_)
+  if [[ "${target}" == powerpc64le-linux-gnu ]]; then
+    OPENBLAS+=(-lgomp)
+  fi
+
+  syms=(DGEMM IDAMAX ISAMAX SNRM2 XERBLA ccopy cgemm cgemv cgeru cher cscal cswap ctrsm ctrsv dcopy dgemm dgemv dger dnrm2 dscal dswap dtrsm dtrsv dznrm2 idamax isamax scnrm2 scopy sgemm sgemv sger snrm2 sscal sswap strsm strsv xerbla zcopy zgemm zgemv zgeru zscal zswap ztrsm ztrsv)
+  for sym in ${syms[@]}
+  do
+    FFLAGS+=("-D${sym}=${sym}_64")
+    CFLAGS+=("-D${sym}=${sym}_64")
+  done
+fi
+
+makefile="Makefile.G95.SEQ"
+cp Make.inc/${makefile} Makefile.inc
+
+make_args+=(OPTF=-O
+            CDEFS=-DAdd_
+            LMETISDIR=/workspace/destdir/lib
+            IMETIS=-I/workspace/destdir/include
+            LMETIS='-L$(LMETISDIR) -lmetis'
+            ORDERINGSF="-Dpord -Dmetis"
+            CC="$CC -fPIC ${CFLAGS[@]}"
+            FC="gfortran -fPIC ${FFLAGS[@]}"
+            FL="gfortran -fPIC"
+            LIBBLAS=${OPENBLAS})
+
+if [[ "${target}" == *-apple* ]]; then
+  make_args+=(RANLIB=echo)
+fi
+
+# NB: parallel build fails
+make alllib "${make_args[@]}"
+
+# build shared libs
+all_load="--whole-archive"
+noall_load="--no-whole-archive"
+extra=""
+if [[ "${target}" == *-apple-* ]]; then
+    all_load="-all_load"
+    noall_load="-noall_load"
+    extra="-Wl,-undefined -Wl,dynamic_lookup -headerpad_max_install_names"
+fi
+
+cd lib
+libs=(-lmetis ${OPENBLAS})
+gfortran -fPIC -shared -Wl,${all_load} libpord.a ${libs[@]} -Wl,${noall_load} ${extra[@]} -o libpord.${dlext}
+cp libpord.${dlext} ${libdir}
+
+libs+=(-L${libdir} -lpord)
+gfortran -fPIC -shared -Wl,${all_load} libmumps_common.a ${libs[@]} -Wl,${noall_load} ${extra[@]} -o libmumps_common.${dlext}
+cp libmumps_common.${dlext} ${libdir}
+
+libs+=(-lmumps_common)
+for libname in cmumps dmumps smumps zmumps
+do
+  gfortran -fPIC -shared -Wl,${all_load} lib${libname}.a ${libs[@]} -Wl,${noall_load} ${extra[@]} -o lib${libname}.${dlext}
+done
+cp *.${dlext} ${libdir}
+cd ..
+
+cp include/* ${prefix}/include
+"""
+
+platforms = expand_gfortran_versions(filter!(p -> !isa(p, Windows) && arch(p) != :powerpc64le, supported_platforms()))
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libsmumps", :libsmumps),
+    LibraryProduct("libdmumps", :libdmumps),
+    LibraryProduct("libcmumps", :libcmumps),
+    LibraryProduct("libzmumps", :libzmumps),
+]
```

Comment:
I think these names are the same as the parallel package.

Reply:
Isn't it a problem that they define the same symbols though?

Reply:
Apparently not. @giordano @staticfloat Can you guys correct me if I am wrong?

Reply:
As it is now, I think `MUMPS_jll` and `MUMPS_seq_jll` won't be able to coexist in the same environment.  On the other hand mangling the name of the library file adds more complexity for all the libraries that will need to link against this one in particular, see the mess we always need to do for OpenBLAS.